QCircLearning/trainerModel.py

```diff
@@ -127,18 +112,21 @@ def NN_opt(func, x0, callback=None, **kwargs):
     sample_y = [func(para) for para in sample_x]
     optimal = [sample_x[np.argmin(sample_y)], np.min(sample_y)]
 
-    print("Training with the neural networks")
-    sys.stdout.flush()
+    if verbose:
+        print("Training with the neural networks")
+        sys.stdout.flush()
 
     for model in nn_models:
-        print(model)
-        sys.stdout.flush()
+        if verbose:
+            print(model)
+            sys.stdout.flush()
```

Comment:
Maybe we can elaborate more on what is going on for each iteration. For example we can print something like `'starting training with model:\n{}'`